### PR TITLE
Git workflow

### PR DESCRIPTION
**Pull request by igorkorsukov**
See https://github.com/igorkorsukov/MuseScoreDocumentation/blob/git_workflow/WorkflowAndGuidelines/GitWorkflow.md

### CODE REVIEW

File: WorkflowAndGuidelines/GitWorkflow.md [hunk 1]
```diff
@@ -1 +1,37 @@
...
+8. Commit changes to your local issue branch: `$ git add . ; git commit -m 'commit message'`
+9. Fetch (download) upstream's master branch: `$ git fetch musescore master`
+10. Update local master to match upstream's master: `$ git checkout master; git rebase musescore/master`
+11. Rebase issue branch: `$ git checkout 1234_new_feature; git rebase master`
+12. Repeat steps 7-11 until dev is complete
+13. Push branch to GitHub: `$ git push origin 1234_new_feature`
+14. Start your browser, go to your GitHub repo, switch to "1234_new_feature" branch and press the [Pull Request] button
+
+After having made a Pull Request don't pull/merge anymore, it'll mess up the commit history. If you (have to) rebase, use `$ git push --force` to send it up to your GitHub repository, this will update the PR too.  
+Be careful not to do this while the core team is working on merging in your PR. Note (to non-UNIX users): the `$ ` of the commands mentioned throughout this page is (meant to be) the shell prompt and as such not to be typed in, your actual shell prompt may look different.
```
Comment (vpereverzev) [inline WorkflowAndGuidelines/GitWorkflow.md:37]: it’s not entirely clear how a person will know that

> Be careful not to do this while the core team is working on merging in your PR

File: WorkflowAndGuidelines/GitWorkflow.md [hunk 2]
```diff
@@ -1 +1,37 @@
 # Git workflow
+
+The development of MuseScore uses GitHub to host the code and manage the collaboration of developers. GitHub is based on git, a popular SCM, initially designed and developed by Linus Torvalds for Linux kernel development. If you're a developer who wants to work on the MuseScore source code and submit your changes to be merged into the main code, here's how.
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply WorkflowAndGuidelines/GitWorkflow.md]: removed

File: WorkflowAndGuidelines/GitWorkflow.md [hunk 3]
```diff
@@ -1 +1,37 @@
...
+
+## Git references
+
+* [Got 15 minutes and want to learn Git?](http://try.github.com)
+* [Fork a repo at GitHub](https://help.github.com/articles/fork-a-repo) - ,most of the text on this page is taken from here.
```
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply WorkflowAndGuidelines/GitWorkflow.md]: thanks, removed

File: WorkflowAndGuidelines/GitWorkflow.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (igorkorsukov) [reply WorkflowAndGuidelines/GitWorkflow.md]: removed

**Pull request merged**